### ISSUE TITLE
terminal accessible buffer, accessible view (views and help menus) don't indicate that they're readonly

### ISSUE THREAD

**Issue by jvesouza** [microsoft/vscode#187774]
Type: <b>Feature Request</b>
@meganrogge @isidorn 

Sometimes, when using the terminal, I accidentally end up typing a command in the Focus Accessible Buffer. Many times I only realize the mistake when pressing the enter key to execute the command. In my opinion, it would be very useful if VSCode issued an alert informing that the focus is on an element that cannot receive data input.

VS Code version: Code - Insiders 1.81.0-insider (d568855b2a4ef75716cdaa9a52da5315f5100a5d, 2023-07-12T11:46:48.659Z)
OS version: Linux x64 6.3.12-200.fc38.x86_64
Modes:

**Comment by isidorn** [comment]
@meganrogge I see that the accessibility help is readonly.

So we could tackle this as a broader request to notify users when they are typing in a readonly editor. This could be an audio cue - windows has some sounds like this where you are typing into a field that does not support it.

**Comment by meganrogge** [comment]
Thanks for the idea @jvesouza this makes a lot of sense to me. Would you rather an alert or audio cue? My thought would be an audio cue as @isidorn suggests. The downside is I think we'd need a new audio cue, which seems to take awhile to get from our sound designer.

Also cc @rperez030 and @jooyoungseo for their thoughts

**Comment by jvesouza** [comment]
In my opinion the audio cue would be better, but an alert will also do. 
There is already an alert message when I open a file, use the File: Toggle Active Editor Read-only command and try to edit something.

**Comment by rperez030** [comment]
What about moving focus to the input field instead? Unless I type a key accidentally, this could even be helpful for productivity. One of the reasons why I still prefer to use the external terminal is because I don't have to wander where my focus is if I want to issue a command.

**Comment by jvesouza** [comment]
> What about moving focus to the input field instead? Unless I type a key accidentally, this could even be helpful for productivity. One of the reasons why I still prefer to use the external terminal is because I don't have to wander where my focus is if I want to issue a command.

In my opinion moving focus to the input field is also a good idea.

One of the advantages for me in using the accessible buffer is how easy it is to navigate the output of commands 
executed in the terminal. I can even copy the result of the command to paste it in a file or send it to someone else.

**Comment by rperez030** [comment]
I agree, and that's why I'd like to use it more. Moving focus automatically if I try to type could be another step in that direction. I'm slightly concerned about accidental typing though. Even in that case, it is still posible to press shift +Tab and the cursor would still be in the same place in the buffer. @jooyoungseo I would be curious to know your thoughts about this.

**Comment by rperez030** [comment]
If we play an audio cue, the accessible buffer should stop accepting characters. It does now if one tries to type in it.

I have also observed that if one moves between the accessible buffer and the input field, the cursor position in the accessible buffer gets reset even if no new command is issued. That is a problem if I'm trying to type a new command and need to review output from the previous one because each time I move focus to the accessible buffer, I have to find my place again. I'm happy to open a separate issue about that.

**Comment by meganrogge** [comment]
The buffer doesn't actually accept characters - the screen reader just reads the characters. I think maybe using `e.preventDefault` would fix this? I'll investigate. 

Also, we need it to accept some characters - arrow keys for instance.

**Comment by rperez030** [comment]
If I type characters, they are not only announced when I type them, they are present if I read the current line, and present in my Braille display. Of course, if I press any arrow key and the cursor moves, they get deleted immediately. Actually, if instead of pressing arrow keys alone I press CTRL +Left or Right arrow keys, I can move word by word and the characters I entered remain as part of the content. It doesn't seem to cause any harm in general, although I can imagine very special situations in which that behavior can be confusing. For example, if I'm reviewing a complicated argument as part of the help information for a command, if I accidentally press a key before reading the text in braille, I will find that ghost character written and may confuse it as being part of the argument.

**Comment by meganrogge** [comment]
Interesting and thanks for that insight. 

Does this happen in all read only editors currently?

**Comment by meganrogge** [comment]
So this doesn't happen in other read only editors currently. the fix is to apply the same contribution that those use.

**Comment by meganrogge** [comment]
Ok interesting - it doesn't happen in the other read only editors. I debugged and realized this is because one of the contributions we provide seems to cause the readonly contribution to break.

**Comment by meganrogge** [comment]
if I provide any contributions at all, it doesn't work

**Comment by meganrogge** [comment]
we should also have this readonly behavior for the `AccessibleView`

**Comment by rperez030** [comment]
The same behavior can be observed in the accessibility help menu.

**Comment by meganrogge** [comment]
yeah that's using the `AccessibleView`

**Comment by rperez030** [comment]
Good to know. I thought the accessible view was something else. :) Thank you for clarifying.
